sbi/inference/potentials/score_fn_iid.py

```diff
@@ -76,12 +78,19 @@ def __init__(
         this requires some adjustments.
 
         Args:
-            score_estimator: The neural network modeling the score.
+            vector_field_estimator: The neural network modeling the score.
             prior: The prior distribution.
             device: The device on which to evaluate the potential. Defaults to "cpu".
         """
 
-        self.score_estimator = score_estimator.to(device).eval()
+        if not vector_field_estimator.SCORE_DEFINED:
+            raise ValueError("Score is not defined for this vector field estimator.")
+        if not vector_field_estimator.MARGINALS_DEFINED:
+            raise ValueError(
+                "Marginals are not defined for this vector field estimator."
```

Comment:
good to have this. can you please add a bit more detail in the error message, e.g., what could be the reason? 
I guess one reason could be that the flow matching based training does not support it, e.g, when using optimal transport flows?